### PR TITLE
Fix #243 - Offset post times to UTC+0 for Calendars

### LINKED ISSUE #243: iCalendar feed doesn't use gmt date for start/end
Publish times are reported incorrectly when subscribing to the iCalendar feed in Google Calendar or Apple Calendar.

Times appear to be adjusted by both the local time on the server and local time in the user's calendar.  iCalendar start/end times should use UTC times.

### PR DESCRIPTION
**Pull request by justnorris**
This fixes a problem where incorrect times were added to calendars because calendars expect a times to be in UTC+0 time zone. The problem described in #243 

Introduced `ics_format_time` static method that is formatting the time properly.

### CODE REVIEW

**Comment by justnorris** [comment]
Removed spacing adjustments for a cleaner PR

File: modules/calendar/calendar.php [hunk 1]
```diff
@@ -489,6 +489,33 @@ function do_ics_escaping( $text ) {
...
+
+		// Timestamp it
+		$timestamp = strtotime( $time_string );
+
+		if( ! $timestamp ) {
+			return false;
+		}
+
+		// Add GMT Offset
+		$timestamp += -1 * get_option('gmt_offset') * HOUR_IN_SECONDS;
```
Comment (sboisvert) [inline modules/calendar/calendar.php]: I'm not sure I understand why we're doing += -1 *, could we not do -= instead?

File: modules/calendar/calendar.php [hunk 2]
```diff
@@ -489,6 +489,33 @@ function do_ics_escaping( $text ) {
...
+
+		// Add manual offset
+		$timestamp += $offset_in_seconds;
+
+		// \T and \Z are escaped for literal T and Z characters
+		return date( 'Ymd\THis\Z', $timestamp );
```
Comment (sboisvert) [inline modules/calendar/calendar.php]: This doesn't seem to be returning a timestamp like the function declaration says it is.

**Comment by justnorris** [comment]
Cleaned up the method comment and the GMT calculation math. Everything should be dandy now :)

**Pull request merged**